### DISCUSSION TITLE
Does Prefect 2.0 Cloud allow hosted database access and remember storage?

### DISCUSSION THREAD

**Discussion by edesz** [PrefectHQ/prefect#5737] [Q&A] [answered]
I have a couple of conceptual questions about using Prefect Cloud with Prefect 2.0, specifically related to the Orion database (to track flow and task runs) and to whether Prefect Cloud can remember my storage configuration.

**Additional Note**
I would only be running flows manually (using scripts) or interactively, and not through deployments.

**Database**

From [this link in the Orion docs](https://orion-docs.prefect.io/ui/overview/#prefect-cloud), it mentions that Prefect Cloud offers
> A hosted Prefect database that stores flow and task run history

When running locally with a SQLite or (local) Postgres database, we can query the tables in the local database. However, when using Prefect Cloud, if we want to access this database and query the tables ourselves, is the (Prefect Cloud)-hosted database accessible? i.e. can we also query tables in the database hosted on Prefect Cloud? In the future, will this be possible?

**Storage**
As (mentioned in the Orion docs [here](https://orion-docs.prefect.io/tutorials/kubernetes-flow-runner/#configure-storage))
> Storage is configured on the Orion server

If I configure storage once in a environment where I am running flows, log in to Prefect Cloud (using `prefect cloud login ...`) and then delete/reset my local environment, does Prefect Cloud still retain knowledge of the storage I previously set?

By this I mean my workflow would be the following
1. I created a Python `virtualenv` with Prefect Orion installed, an am running Prefect 2.0 flows locally inside that environment
2. I configured storage from inside that `virtualenv` using
   ```bash
   prefect storage create
   ```
3. still from inside the same `virtualenv`, I now log in to Prefect Cloud using
   ```bash
   prefect cloud login ...
   ```
4. finally, I delete
   - this `virtualenv` and create a new one
   - everything in `~/.prefect/storage`
5. (note) I have not run `prefect orion start` at all
6. I now create a **new** Python `virtualenv` and run
   ```bash
   prefect storage ls
   ```

Will the output of step 6. in this workflow list the storage I had (previously) configured in the deleted `virtualenv`? Does Prefect Cloud have any way to remember this storage that I had configured?

Rather than virtual environments, this could be different host systems
- eg. I'm running on workstation 1 today and configure storage for Prefect Cloud, the OS on the workstation (eg. Ubuntu Server) is then wiped (the Orion installation is obviously gone) and I'm now running on workstation 2. In this scenario, do I need to re-configure **new** storage for Prefect Cloud? Or can I somehow re-use the old one?

Is there a way to configure storage with Prefect Cloud once and then just re-use it in the future?

**Comment by zanieb** [comment, marked as answer]
When using Cloud, you cannot access the database directly. We intend to provide REST routes that should address most use-cases, including analytics focused ones.

Storage settings are persisted to the database and do not automatically transfer between a local API and the Cloud API. We intend to make it easy to transfer storage and deployments from your local API to the Cloud API though.

**Comment by edesz** [comment]
Thanks @madkinsz. That sounds great. Especially for the second part, having a convenient approach to re-use storage will be quite helpful.

Prefect 2.0 Cloud is looking good so far. It is quite nice.
